### PR TITLE
Fixes issue #430

### PR DESCRIPTION
**Pull request by WSSDude**

### CODE REVIEW

File: src/doom/r_data.c [hunk 1]
```diff
@@ -376,7 +376,7 @@ void R_GenerateComposite (int texnum)
 
     // Now that the texture has been built in column cache,
     //  it is purgable from zone memory.
-    Z_ChangeTag (block, PU_CACHE);
```
Comment (fabiangreffrath) [inline src/doom/r_data.c]: This one is wrong. The memory `block` points to has been `Z_Malloc()`ed, so there is no lump to be released. Please revert back to the original code.

File: src/doom/p_switch.c [hunk 2]
```diff
@@ -25,7 +25,7 @@
 #include "p_local.h"
 #include "i_swap.h" // [crispy] SHORT()
 #include "w_wad.h" // [crispy] W_CheckNumForName()
-#include "z_zone.h" // [crispy] Z_ChangeTag()
+#include "z_zone.h" // [crispy] purge tags
```
Comment (fabiangreffrath) [inline src/doom/p_switch.c]: Strictly speaking, we only include this for the `PU_STATIC` in `W_CacheLumpName("SWITCHES", PU_STATIC)`. So, `PU_STATIC` would be the right comment.

**Comment by WSSDude** [comment]
> 
> 
> Last nit-pick, please provide a sensible subject line and move the `fixes ...` stanza into the longer description.

Don't quite understand what you mean by this... Other changes should be on their way. Btw, if Z_Malloc is used, Z_ChangeTag needs to be utilized if I understood correctly... This is only for cases where W_CacheLumpNum/Name is used, right?

**Comment by fabiangreffrath** [comment]
> Don't quite understand what you mean by this... 

You called the commit "Fixes issue #xyz", but this doesn't tell what you actually change in this commit. You have to know the issue to understand what this code change is about. I suggest to call the commit "avoid use of Z_ChangeTag() for memory allocated by W_CacheLumpNum()" or similar and move the information that this fixes a specific issue further down into the commit description.

> Other changes should be on their way. Btw, if Z_Malloc is used, Z_ChangeTag needs to be utilized if I understood correctly... This is only for cases where W_CacheLumpNum/Name is used, right?

Yes, this is only about memory allocated by W_CacheLumpNum/Name, i.e. for lump content copied to memory.

**Comment by WSSDude** [comment]
Ah ok. I'll make sure to be more descriptive next time.

**Pull request merged**